packages/ckeditor5-engine/src/model/utils/insertcontent.ts

```diff
@@ -77,18 +78,111 @@ export default function insertContent(
 		}
 
 		const insertion = new Insertion( model, writer, selection.anchor! );
+		const markersPosition = [];
+		const fakeMarkerElements = [];
 
 		let nodesToInsert: any;
 
 		if ( content.is( 'documentFragment' ) ) {
+			if ( content.markers.size ) {
+				for ( const [ name, range ] of content.markers ) {
+					const { start, end } = range;
+					const isCollapsed = start.isEqual( end );
+
+					markersPosition.push( {
+						position: start,
+						name,
+						isCollapsed
+					}, {
+						position: end,
+						name,
+						isCollapsed
+					} );
+				}
+
+				markersPosition.sort( ( { position: posA }, { position: posB } ) => posA.isBefore( posB ) ? 1 : -1 );
+
+				for ( const { position, name, isCollapsed } of markersPosition ) {
+					let fakeElement = null;
+					let collapsed = null;
+					const isBeginning = position.parent === content && position.isAtStart;
+					const isEnding = position.parent === content && position.isAtEnd;
+
+					if ( !isBeginning && !isEnding ) {
+						fakeElement = writer.createElement( '$marker', { 'data-name': name } );
```

Comment:
Since this is not an HTML element, you can simply use "name" as the attribute ... name. Nothing bad should happen.

Reply:
BTW. it seems that you don't even read the name stored in this attribute :D.